nomos-da/network/subnetworks-assignations/src/lib.rs

```diff
@@ -0,0 +1,11 @@
+use std::collections::HashSet;
+use std::hash::Hash;
+
+pub trait MembershipHandler {
```

Comment:
As of now I feel that this trait would better fit inside the `network/core/membership/mod.rs`. The actual functions to calculate the subnetwork idx from the pov key could also live there (assuming `nomos-da` crates are considered as a libraries).

The composition of these parts will happen in `nomos-services`, where we'll be able to provide different adapters (eg. `MockMemebershipAdapter`?

Reply:
Yeah, the trait would fit there as well. But I think as those algorithms are "simple" but will change in the future they deserved their own crate. Then in the nomos-services we just compose them indeed.

Reply:
🤝